src/generator/resources/Vector.st

```diff
@@ -47,6 +53,91 @@ public class $vector/mutableType()$ extends $vector/baseType()$ {
         this($components:{x| other.get$x.proper$()}; separator = ", "$);
     }
 
+$if(componentType.discrete)$
+    /**
+     $components:{x| * @param $x$ the $x$ component}; separator = "\n"$
+     */
+    public $vector/mutableType()$($components:{x|float $x$}; separator = ", "$) {
+        this($components:{x|
+            DoubleMath.roundToInt($x$, RoundingMode.FLOOR)};
```

Comment:
Does this need to be done for this PR?   All seems to be working well in the engine with these changes.